oxenss/snode/service_node.cpp

```diff
@@ -678,28 +722,11 @@ void ServiceNode::update_swarms() {
 
                         // Request some recent block hash heights so that we can properly carry out
                         // and respond to storage testing (for which we need to know recent block
-                        // hashes). Incoming tests are *usually* height - TEST_BLOCKS_BUFFER, but
-                        // request a couple extra as a buffer.
-                        for (uint64_t h = bu.height - TEST_BLOCKS_BUFFER - 2; h < bu.height; h++)
-                            omq_server_.oxend_request(
-                                    "rpc.get_block_hash",
-                                    [this, h](bool success, std::vector<std::string> data) {
-                                        if (!(success && data.size() == 2 && data[0] == "200" &&
-                                              data[1].size() == 66 && data[1].front() == '"' &&
-                                              data[1].back() == '"'))
-                                            return;
-                                        std::string_view hash{
-                                                data[1].data() + 1, data[1].size() - 2};
-                                        if (oxenc::is_hex(hash)) {
-                                            log::debug(
-                                                    logcat,
-                                                    "Pre-loaded hash {} for height {}",
-                                                    hash,
-                                                    h);
-                                            block_hashes_cache_.insert_or_assign(h, hash);
-                                        }
-                                    },
-                                    "{\"height\":[" + util::int_to_string(h) + "]}");
+                        // hashes). There's a race condition here with `on_swarm_update` both
+                        // writing to `block_hashes_cache` hence writes to it need to be
+                        // synchronised which we do here by blocking.
+                        blocking_request_for_recent_block_hashes(
+                                bu.height, omq_server_, block_hashes_cache_);
```

Comment:
This whole thing can actually just get deleted: peer testing has not been done in a couple of years and is basically just dead code right now.

It never actually worked reliably with many false positives, never actually had any consequence, and was a source of frequent spurious log error spam.  It was also essentially impossible to make it have consequences, because if you fail for some reason (e.g. you are missing a message) there is no actual way to fix it.

Reply:
Outdated comment on earlier commit.